### PR TITLE
Add poll

### PR DESCRIPTION
**Pull request by AlicePru**
### Checklist
- [x] I've run `mvn test` from the root directory to see all new and existing tests pass
- [x] I've followed code style and run and ensured the code style is valid
- [x] I've created new tests if necessary.

### Motivation and Context
#15 
#16 
#14 
#41 
#22 
#23 
#24 
#50

### Description
Were added opportunity to create poll and display feedback and voting in dependency of event status(open or closed) . 
P.S. Not all methods covered by Unit Tests

### CODE REVIEW

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 1]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
 
         return event.stream()
-                .map(e -> new EventDto(e.getName(), e.getDescription(), e.getDate(), e.getTime(), e.getId(), e.getAgenda()))
+                .map(e -> new EventDto(e.getName(), e.getDescription(), e.getDate(), e.getTime(), e.getId(), e.getAgenda(), e.getStatus()))
                 .collect(Collectors.toList());
     }
 
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/savePoll/{id}")
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: change url

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 2]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+            answer.setText(answersDto.getText());
+            answerList.add(answer);
+        }
+        poll.setAnswers(answerList);
+        em.persist(poll);
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/getPoll/{id}")
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: change url

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 3]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+                .collect(Collectors.toList());
+    }
+
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/vote/{id}")
+    public void vote(@PathParam("id") Long id){
+        Optional<Answer> answer=answersStore.getAnswerByID(id);
+        if(answer.isPresent()){
+            Answer a=answer.get();
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Change variable name, method name .. Why counter doesn't persists anywhere ?

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 4]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+            a.setCounter(a.getCounter()+1);
+            System.out.print(a.getCounter());
+        }
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/getVotes/{id}")
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: change url

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 5]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+    public List<AnswersDto> getVotes(@PathParam("id") Long id){
+        Poll poll=new Poll();
+        poll.setId(id);
+        List<Answer> answerList=answersStore.getAnswersByPollID(poll);
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Why artificial poll object is created ?

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 6]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+        System.out.print(answerList);
+        return answerList.stream()
+                .map(a->new AnswersDto(a.getId(),a.getCounter()))
+                .collect(Collectors.toList());
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/getFeedbackPoll/{id}")
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: check if id = eventId and change url

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 7]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+                        toAnswersDtoList(p.getAnswers()),
+                        p.isFeedback(),
+                        p.getEventID(),
+                        p.getId()))
+                .collect(Collectors.toList());
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/getVotingPoll/{id}")
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: same as feedback poll

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 8]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+        return poll.stream()
+                .map(p -> new PollDto(p.getQuestion(),
+                        toAnswersDtoList(p.getAnswers()),
+                        p.isFeedback(),
+                        p.getEventID(),
+                        p.getId()))
+                .collect(Collectors.toList());
+    }
+
+    private List<AnswersDto> toAnswersDtoList(List<Answer> list) {
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Move somewhere from /API class

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 9]
```diff
@@ -80,17 +88,125 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+        return list.stream()
+                .map(this::toDto)
+                .collect(Collectors.toList());
+    }
+
+    private AnswersDto toDto(Answer answer) {
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Move somewhere from /API class

File: src/main/java/lv/ctco/javaschool/eventorganaizer/entity/Event.java [hunk 10]
```diff
@@ -29,27 +29,27 @@
 
     private EventStatus status;
 
-    public Event(String name, User author, String description, String agenda, String date, String time) {
```
Comment (StepanovsRomans) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/entity/Event.java]: format!

File: src/main/webapp/app/jsp/add-event.jsp [hunk 11]
```diff
@@ -46,72 +48,92 @@
...
-            document.getElementById("save").classList.remove("w3-hide");
-        }
-    }
 
-    function saveDataToDB() {
-        getDataFromField();
-            if(data.name == "" || data.name == " ")  {
-            alert("Please input Event Name");
-            return;
+        function saveDataToDB() {
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/add-event.jsp]: refactor function
validation should be in different function

File: src/main/webapp/app/jsp/add-event.jsp [hunk 12]
```diff
@@ -129,57 +151,56 @@
...
-            return response.json();
-        }).then(function (event) {
-            document.getElementById("name").value = event.eventName;
-            document.getElementById("description").value = event.eventDescription;
-            document.getElementById("datepicker").value = event.eventDate;
-            document.getElementById("timepicker").value = event.eventTime;
-            document.getElementById("agenda").value = event.eventAgenda;
-        })
-    }
+        function getEventDataFromDB() {
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/add-event.jsp]: is it form DB ?

File: src/main/webapp/app/jsp/add-event.jsp [hunk 13]
```diff
@@ -129,57 +151,56 @@
...
+        }
 
-    function getQueryVariable(variable) {
-        var query = window.location.search.substring(1);
-        var vars = query.split("&");
-        for (var i = 0; i < vars.length; i++) {
-            var pair = vars[i].split("=");
-            if (pair[0] == variable) {
-                return pair[1];
+        function getQueryVariable(variable) {
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/add-event.jsp]: variable names ?

Comment (StepanovsRomans) [reply src/main/webapp/app/jsp/add-event.jsp]: check why it is used and remove if there is different way to get id.

File: src/main/webapp/app/jsp/event.jsp [hunk 14]
```diff
@@ -42,12 +73,108 @@
                 w3.displayObject("title", event);
                 w3.displayObject("event-field", event);
             }
+            if (event.eventStatus === "OPEN") {
+                document.getElementById("voting").classList.remove("w3-hide");
+                document.getElementById("feedback").classList.add("w3-hide");
+                getVotingFromDB();
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/event.jsp]: DB ?

File: src/main/webapp/app/jsp/event.jsp [hunk 15]
```diff
@@ -42,12 +73,108 @@
...
+            }
+            if (event.eventStatus === "CLOSED") {
+                document.getElementById("voting").classList.add("w3-hide");
+                document.getElementById("feedback").classList.remove("w3-hide");
+                getFeedbackFromDB()
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/event.jsp]: DB ?

File: src/main/webapp/app/jsp/poll.jsp [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+        <button onclick="deletePoll('{{id}}')">Delete Poll</button>
+        <hr/>
+    </div>
+    {{/pollArray}}
+</script>
+
+<div id="displayPoll">
+</div>
+
+<script>
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/poll.jsp]: scripts in different files

File: src/main/webapp/app/jsp/poll.jsp [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+        }).then(function (event) {
+            console.log(JSON.stringify(event));
+            if (event !== undefined) {
+                document.getElementById("event-field").classList.remove("w3-hide");
+                w3.displayObject("event-field", event);
+            }
+        })
+    }
+
+    function getQueryVariable(variable) {
```
Comment (StepanovsRomans) [inline src/main/webapp/app/jsp/poll.jsp]: code dublicate

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 18]
```diff
@@ -32,6 +41,13 @@
     @Inject
     private EventStore eventStore;
 
+    @Inject
+    private PollStore pollStore;
+
+    @Inject
+    private AnswersStore answersStore;
+
+
```
Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Please remove extra blank line

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 19]
```diff
@@ -67,30 +82,141 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+                        p.getId()))
+                .collect(Collectors.toList());
+    }
+
+    @GET
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/{id}/getVotingPoll")
+    public List<PollDto> getVotingForEvent(@PathParam("id") Long id) {
+        List<Poll> poll = pollStore.getVotingPoll(id);
+        return poll.stream()
```
Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Please remove duplication by creating a mapPollToDto(poll) method

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 20]
```diff
@@ -67,30 +82,141 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+
+        List<Answer> answerList = new ArrayList<>();
+        for (AnswerDto answerDto : answersString) {
+            Answer answer = new Answer();
+            answer.setPoll(poll);
+            answer.setText(answerDto.getText());
+            answerList.add(answer);
+        }
+        poll.setAnswers(answerList);
+        em.persist(poll);
```
Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Please put db logic into corresponding Store

File: src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java [hunk 21]
```diff
@@ -67,30 +82,141 @@ public EventDto getEventById(@PathParam("id") Long id) throws IllegalArgumentExc
...
+                .map(e -> new EventDto(e.getName(), e.getDescription(), e.getDate(), e.getTime(), e.getId(), e.getAgenda(), e.getStatus()))
                 .collect(Collectors.toList());
     }
 
+    @POST
+    @RolesAllowed({"ADMIN", "USER"})
+    @Path("/{id}/savePoll")
+    public void savePoll(PollDto pollDto, @PathParam("id") Long id) {
+        List<AnswerDto> answersString = pollDto.getAnswers();
+        //String[] answerList = answersString.split("\n");
```
Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Please remove comment if it is not needed

Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/boundary/EventOrganizationApi.java]: Do not commit comments please

File: src/main/java/lv/ctco/javaschool/eventorganaizer/control/AnswersStore.java [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+
+@Stateless
+public class AnswersStore {
+    @PersistenceContext
+    private EntityManager em;
+
+    @Inject
+    private AnswersStore answersStore;
+
+
```
Comment (AlexeyBuzdin) [inline src/main/java/lv/ctco/javaschool/eventorganaizer/control/AnswersStore.java]: Not needed extra line

**Pull request merged**